tests/foreman/api/test_setting.py

```diff
@@ -18,228 +18,187 @@
+    setting_update.value = login_text_value
+    setting_update.update({'value'})
+    assert setting_update.value == login_text_value
+
+
+@run_in_one_thread
+@tier2
+@pytest.mark.parametrize('setting_update', ['login_text'], indirect=True)
+def test_positive_update_login_page_footer_text_without_value(setting_update):
+    """Updates parameter "login_text" without any string (empty value)
+
+    :id: 7a56f194-8bde-4dbf-9993-62eb6ab10733
+
+    :expectedresults: login_text has empty value after update
+
+    :parametrized: yes
+    """
+    setting_update.value = ""
+    setting_update.update({'value'})
+    assert setting_update.value == ""
+
+
+@run_in_one_thread
+@tier2
+@pytest.mark.parametrize('setting_update', ['login_text'], indirect=True)
+def test_positive_update_login_page_footer_text_with_long_string(setting_update):
+    """Attempt to update parameter "Login_page_footer_text"
+        with long length string
+
+    :id: fb8b0bf1-b475-435a-926b-861aa18d31f1
+
+    :expectedresults: Parameter is updated
+
+    :parametrized: yes
+    """
+    login_text_value = [login_text for login_text in generate_strings_list(1000)][0]
+    setting_update.value = login_text_value
+    setting_update.update({'value'})
+    assert setting_update.value == login_text_value
+
+
+@pytest.mark.skip_if_open("BZ:1470083")
+@tier2
+@pytest.mark.parametrize('setting_update', ['discovery_hostname'], indirect=True)
+def test_negative_update_hostname_with_empty_fact(setting_update):
+    """Update the Hostname_facts settings without any string(empty values)
+
+    :id: b8e260fc-e263-4292-aa2f-ab37085c7758
+
+    :expectedresults: Error should be raised on setting empty value for
+        hostname_facts setting
+
+    :parametrized: yes
+    """
+    setting_update.value = ""
+    setting_update.update({'value'})
+    assert setting_update.value != "", "Empty string"
+
+
+@tier2
+@pytest.mark.parametrize('setting_update', ['discovery_prefix'], indirect=True)
+def test_positive_update_hostname_prefix_without_value(setting_update):
+    """Update the Hostname_prefix settings without any string(empty values)
+
+    :id: 3867488c-d955-47af-ac0d-71f4016391d1
+
+    :expectedresults: Hostname_prefix should be set without any text
+
+    :parametrized: yes
+    """
+    setting_update.value = ""
+    setting_update.update({'value'})
+    assert setting_update.value == ""
+
+
+@tier1
+@pytest.mark.parametrize('setting_update', ['discovery_prefix'], indirect=True)
+def test_positive_update_hostname_default_prefix(setting_update):
+    """Update the default set prefix of hostname_prefix setting
+
+    :id: 4969994d-f934-4f0e-9a98-476b87eb0527
+
+    :CaseImportance: Critical
+
+    :parametrized: yes
+
+    :expectedresults: Default set prefix should be updated with new value
+    """
+    discovery_prefix = [
+        discovery_prefix
+        for discovery_prefix in generate_strings_list(exclude_types=['alphanumeric', 'numeric'])
+    ][0]
```

Comment:
Same comment about discarding tests for all other values

Reply:
It works, generate_strings_list function returns a list of random value.

Reply:
fixed.

Reply:
fixed.